nucliadb/nucliadb/common/datamanagers/resources.py

```diff
@@ -219,3 +225,29 @@ async def modify_slug(txn: Transaction, *, kbid: str, rid: str, new_slug: str) -
 
 async def set_resource_shard_id(txn: Transaction, *, kbid: str, rid: str, shard: str):
     await txn.set(KB_RESOURCE_SHARD.format(kbid=kbid, uuid=rid), shard.encode())
+
+
+async def get_all_field_ids(
+    txn: Transaction, *, kbid: str, rid: str
+) -> Optional[resources_pb2.AllFieldIDs]:
+    key = KB_RESOURCE_ALL_FIELDS.format(kbid=kbid, uuid=rid)
+    return await get_kv_pb(txn, key, resources_pb2.AllFieldIDs)
+
+
+async def set_all_field_ids(
+    txn: Transaction, *, kbid: str, rid: str, allfields: resources_pb2.AllFieldIDs
+):
+    key = KB_RESOURCE_ALL_FIELDS.format(kbid=kbid, uuid=rid)
+    await txn.set(key, allfields.SerializeToString())
+
+
+async def has_field(
+    txn: Transaction, *, kbid: str, rid: str, field_id: resources_pb2.FieldID
+) -> bool:
```

Comment:
Sorry, last comment: It would be nice to have tests for those methods, as they are meant to be used across all other parts of the codebase

Reply:
Added a simple test for these functions. Existing tests cover orm resource accessing and updating this key